mica-webapp/src/main/webapp/app/share-resource/share-resource.js

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *  * Copyright (c) 2017 OBiBa. All rights reserved.
+ *  *
+ *  * This program and the accompanying materials
+ *  * are made available under the terms of the GNU Public License v3.0.
+ *  *
+ *  * You should have received a copy of the GNU General Public License
+ *  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+'use strict';
+
+mica.shareResource = angular.module('mica.shareResource', [
+  'mica.config',
+  'obiba.notification',
+  'obiba.form',
+  'obiba.mica.localized',
+  'pascalprecht.translate',
+  'ui.bootstrap'
+]);
+
+mica.shareResource
+  .factory('ShareService', ['$resource',
+    function ($resource) {
+
+      return $resource('ws/draft/:resourceType/:resourceId/_share', {}, {
+        'getShareLink': {
+          method: 'PUT',
+          params: {resourceType: '@resourceType', resourceId: '@resourceId', expire: '@expire'},
+          responseType: 'text',
+          transformResponse: function (response) {
```

Comment:
Yep, the resource transformed the string into an array of characters rendering the output unusable.